indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -185,3 +188,31 @@ class RHRegistrationEdit(RHManageRegistrationBase):
     def _process(self):
         return WPManageRegistration.render_template('management/registration_modify.html', self.event,
                                                     registration=self.registration)
+
+
+class RHRegistrationEmailRegistrants(RHManageRegFormBase):
+
+    def _checkParams(self, params):
+        self._doNotSanitizeFields.append('from_address')
+        RHManageRegFormBase._checkParams(self, params)
+
+    def _get_people(self):
+        identifiers = set(request.form.getlist('references'))
+        return Registration.find_all(Registration.id.in_(identifiers))
+
+    def _success_handler(self, form):
+        people = self._get_people()
+        for person in people:
+            template = get_template_module('events/registration/emails/custom_email.html',
+                                           email_subject=form.subject.data, email_body=form.body.data)
+            email = make_email(to_list=person.email, cc_list=form.cc_addresses.data,
+                               from_address=form.from_address.data, template=template, html=True)
+            send_email(email, self.event)
+
+    def _process(self):
+        form = EmailRegistrantsForm()
+        if form.validate_on_submit():
+            self._success_handler(form)
+            flash(_("Emails sent"), 'success')
+            return jsonify(success=True)
```

Comment:
`return jsonify_data()` and it'll pass the flashed messages back to the JS automatically (ajaxDialog shows them on the page that opened it).